tools/src/commands/Vendor.ts

```diff
@@ -172,6 +173,146 @@ function getGestureHandlerPipe() {
   );
 }
 
+function getSafeAreaPipe() {
+  const destination = 'packages/expo-dev-menu/vendored/react-native-safe-area-context';
+
+  // prettier-ignore
+  return new Pipe().addSteps(
+    'all',
+      new Clone({
+        url: 'git@github.com:th3rdwave/react-native-safe-area-context.git',
+        tag: 'v3.3.2',
+      }),
+      new RemoveDirectory({
+        name: 'clean vendored folder',
+        target: destination,
+      }),
+      new CopyFiles({
+        filePattern: ['src/**/*.*', '*.d.ts'],
+        to: destination,
+      }),
+    'android',
+      prefixPackage({
+        packageName: 'com.th3rdwave.safeareacontext',
+        prefix: 'devmenu',
+      }),
+      new CopyFiles({
+        subDirectory: 'android/src/main/java/com/th3rdwave',
+        filePattern: '**/*.@(java|kt|xml)',
+        to: path.join(destination, 'android/devmenu/com/th3rdwave'),
+      }),
+      new CopyFiles({
+        subDirectory: 'android/src/main/java/com/facebook',
+        filePattern: '**/*.@(java|kt|xml)',
+        to: path.join(destination, 'android/com/facebook'),
+      }),
+
+    'ios',
+      new TransformFilesName({
+        filePattern: 'ios/**/*RNC*.@(m|h)',
+        find: 'RNC',
+        replace: 'DevMenuRNC',
+      }),
+      new TransformFilesName({
+        filePattern: 'ios/**/*SafeAreaCompat.@(m|h)',
+        find: 'SafeAreaCompat',
+        replace: 'DevMenuSafeAreaCompat',
+      }),
+      renameIOSSymbols({
+        find: 'RNC',
+        replace: 'DevMenuRNC',
+      }),
+      renameIOSSymbols({
+        find: 'SafeAreaCompat',
+        replace: 'DevMenuSafeAreaCompat',
+      }),
+      new TransformFilesContent({
+        filePattern: 'ios/**/*.@(m|h)',
+        find: 'UIEdgeInsetsEqualToEdgeInsetsWithThreshold',
+        replace: 'DevMenuUIEdgeInsetsEqualToEdgeInsetsWithThreshold',
+      }),
+      new TransformFilesContent({
+        filePattern: 'ios/**/DevMenuRNCSafeAreaProviderManager.@(m|h)',
+        find: '^RCT_EXPORT_MODULE\\((.*)\\)',
+        replace: '+ (NSString *)moduleName { return @"RNCSafeAreaProvider"; }',
+      }),
+      new TransformFilesContent({
+        filePattern: 'ios/**/DevMenuRNCSafeAreaViewManager.@(m|h)',
+        find: '^RCT_EXPORT_MODULE\\((.*)\\)',
+        replace: '+ (NSString *)moduleName { return @"RNCSafeAreaView"; }',
+      }),
+      
+      new TransformFilesContent({
```

Comment:
couldn't think of a better way to replace / wrap a method with `dispatch_sync` -- if you know of a better way to do this please let me know!

Reply:
I think it's good. Maybe we could create a better action in the future, but for now, you can leave it as it is.